test/objectid.test.js

```diff
@@ -55,18 +57,19 @@ describe('ObjectID', function() {
     ObjectID(id).should.be.equal(123);
   });
 
-  it('coerces ObjectID', function() {
-    const coercedId = ds.connector.isObjectIDProperty('Book', {}, '52fcef5c0325ace8dcb7a0bd');
-    coercedId.should.be.True();
-  });
-
-  it('given strictObjectIDCoercion: true, does not coerce ObjectID', function() {
-    const coercedId = ds.connector.isObjectIDProperty(
-      'Book',
-      {},
-      '52fcef5c0325ace8dcb7a0bd',
-      {strictObjectIDCoercion: true}
-    );
-    coercedId.should.be.False();
+  context('strictObjectIDCoercion', function() {
+    context('set to true', function() {
+      it('should not coerce to ObjectID', async function() {
```

Comment:
I feel like the tests replaced by your new test are more like unit tests for `isObjectIDProperty`, and your new one is more like an acceptance test (I guess it does call `isObjectIDProperty` behind the scenes). Can we preserve the old tests?

Reply:
Previously `isObjectIDProperty` was a method on the prototype, now it is not. I removed it because I wanted to avoid adding helper functions to the prototype.

If we want to unit test the helper function `isObjectIDProperty`, shouldn't we test the rest of the helper functions too? We should ideally, but it seems like an overkill to me, because these helper functions are not exposed as connector methods and their functionality is getting tested via the integration tests, anyway. And to unit test these helper functions, we will have to add all of them to the prototype; I am not in favor of doing that.

@bajtos thoughts?

Reply:
I see. Thanks for the background context on this @hacksparrow. I understand now it's not a prototype method anymore, but the tests _did_ exist for it, and I'm just asking to preserve them. We don't have to necessarily write unit tests for all the helper methods. I'm okay to not recover those tests too.

Reply:
> I'm just asking to preserve them

@b-admike Preserve, as in "don't get rid of the old test"? If yes, what is the point of keeping it when we already know it will fail, since the test calls a now non-existent method?

I can understand your concern that a unit test is gone, but that particular condition and many more are now checked from a higher level.

The reason why I removed `isObjectIDProperty` from the prototype is because:

1. I wanted to modularize ObjectID-related functionality in the whole module, so had to created smaller helper functions, which are can be used individually too, and adds clarity to the flow.
2. If `isObjectIDProperty` was to be on the prototype, some of the helper functions had to be there too.
3. Removing `isObjectIDProperty` from the prototype was the more elegant approach than adding some more helper functions to the prototype, IMO.

Reply:
Thank you for clarifying @hacksparrow. Ah I thought the tests could be recovered with the changes that took place. I'm good with leaving them out.